service/history/tasks/category.go

```diff
@@ -45,133 +41,76 @@ type (
+	categoryNameArchival    = "archival"
+	categoryNameMemoryTimer = "memory-timer"
 )
 
 var (
 	CategoryTransfer = Category{
 		id:    CategoryIDTransfer,
 		cType: CategoryTypeImmediate,
-		name:  CategoryNameTransfer,
+		name:  categoryNameTransfer,
 	}
 
 	CategoryTimer = Category{
 		id:    CategoryIDTimer,
 		cType: CategoryTypeScheduled,
-		name:  CategoryNameTimer,
+		name:  categoryNameTimer,
 	}
 
 	CategoryReplication = Category{
 		id:    CategoryIDReplication,
 		cType: CategoryTypeImmediate,
-		name:  CategoryNameReplication,
+		name:  categoryNameReplication,
 	}
 
 	CategoryVisibility = Category{
 		id:    CategoryIDVisibility,
 		cType: CategoryTypeImmediate,
-		name:  CategoryNameVisibility,
+		name:  categoryNameVisibility,
 	}
 
 	CategoryArchival = Category{
 		id:    CategoryIDArchival,
 		cType: CategoryTypeScheduled,
-		name:  CategoryNameArchival,
+		name:  categoryNameArchival,
 	}
 
 	CategoryMemoryTimer = Category{
 		id:    CategoryIDMemoryTimer,
 		cType: CategoryTypeScheduled,
-		name:  CategoryNameMemoryTimer,
-	}
-)
-
-var (
-	categories = struct {
-		sync.RWMutex
-		m map[int32]Category
-	}{
-		m: map[int32]Category{
-			CategoryTransfer.ID():    CategoryTransfer,
-			CategoryTimer.ID():       CategoryTimer,
-			CategoryVisibility.ID():  CategoryVisibility,
-			CategoryReplication.ID(): CategoryReplication,
-			CategoryMemoryTimer.ID(): CategoryMemoryTimer,
-		},
+		name:  categoryNameMemoryTimer,
 	}
 )
 
-// NewCategory creates a new Category and register the created Category
-// Registered Categories can be retrieved via GetCategories() or GetCategoryByID()
-// NewCategory panics when a Category with the same ID has already been registered
-func NewCategory(
-	id int32,
-	categoryType CategoryType,
-	name string,
-) Category {
-	categories.Lock()
-	defer categories.Unlock()
-
-	if category, ok := categories.m[id]; ok {
-		panic(fmt.Sprintf("category id: %v has already been defined as type %v and name %v", id, category.cType, category.name))
-	}
-
-	newCategory := Category{
+func NewCategory(id int, cType CategoryType, name string) Category {
+	return Category{
 		id:    id,
-		cType: categoryType,
+		cType: cType,
 		name:  name,
 	}
-	categories.m[id] = newCategory
-
-	return newCategory
-}
-
-// GetCategories returns a deep copy of all registered Categories
-func GetCategories() map[int32]Category {
-	categories.RLock()
-	defer categories.RUnlock()
-	return maps.Clone(categories.m)
-}
-
-// RemoveCategory removes a registered Category.
-// This should only be used for testing.
-func RemoveCategory(id int32) {
-	categories.Lock()
-	defer categories.Unlock()
-	delete(categories.m, id)
-}
-
-// GetCategoryByID returns a registered Category with the same ID
-func GetCategoryByID(id int32) (Category, bool) {
-	categories.RLock()
-	defer categories.RUnlock()
-
-	category, ok := categories.m[id]
-	return category, ok
 }
 
-func (c Category) ID() int32 {
+func (c Category) ID() int {
```

Comment:
For example, doing things like strconv.Itoa will need to convert this to an int. The protos force us to use int32, but, after that, the longer we keep int32 around, the more places we'll have to convert it to an int when we need it as one. It's a similar to argument to why you would want to use `time.Time` instead of the protobuf timestamp. We may overall end up with a higher number of conversions to/from int/int32. However, the use cases will always be the same this way--going to/from protos. Versus if we leave it as an int32, the number of reasons we'll need to convert can explode.